mantidproject/mantid
# Fix fit range for multiple inputs in indirect data analysis

**Description of work.**

This PR fixes an issue with the fitting of multiple spectra in indirect data analysis where it would only use the fit range of the first spectra.

**Report to:** Spencer

**To test:**

1.`Interfaces`->`Indirect`->`Data Analysis`->`ConvFit` tab
2. Go to Multiple Input and load the data below
3. Click Run and wait.
4. Select the OSIRIS workspace using the combobox
5. Note that the fit range of the OSIRIS is the same as that listed in the table.

**Data Files**
[ConvFit Files.zip](https://github.com/mantidproject/mantid/files/2817435/ConvFit.Files.zip)

Fixes #24664 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
